web/app/Actions/Fortify/CreateNewUser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Wikijump\Actions\Fortify;
+
+use Wikijump\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+use Laravel\Fortify\Contracts\CreatesNewUsers;
+
+class CreateNewUser implements CreatesNewUsers
+{
+    use PasswordValidationRules;
+
+    /**
+     * Validate and create a newly registered user.
+     *
+     * @param  array  $input
+     * @return \Wikijump\Models\User
+     */
+    public function create(array $input)
+    {
+        Validator::make($input, [
+            'name' => ['required', 'string', 'max:255'],
```

Comment:
I'm not remotely married to the language styling, it is expected in a few places in the code but nothing that can't be modified. You may have to show me what you have in mind and we'll make it work.

Reply:
I'm not sure what you mean by "language styling". I just meant that usernames are required to be at least two or three characters, so a username like `Z` isn't valid. Presumably you can do that here by just `min:3`?

Reply:
Sorry, for whatever reason GitHub's not letting me comment on the actual thing I was trying to respond to which was `en` for `DEFAULT_LANGUAGE`.

Reply:
Oh alright, I see.

Reply:
So for the language stuff, I meant that the default language has debug strings. So instead of
```php
_("This user does not exist! ;-)")
```
You have
```php
_("error-user-not-exist")
```
So any areas where we would otherwise have hardcoded English / the language parameter is missing, you'll see `error-user-not-exist` on the web page, which is clearly an error.

I want to be vigorous about avoiding hardcoded English, since Wikidot plays things pretty loose and the experience for international users is pretty inconsistent and atrocious, especially if they aren't fluent in English.

Reply:
My only reservation with that pattern is that the locale translation file is _much_ easier for a translator to read when they have the original English meaning side-by-side with their translation. See https://laravel.com/docs/8.x/localization#using-translation-strings-as-keys

Reply:
That's true. I guess one of my concerns is that when you update the English copy you have to update all the translation files as well, something Wikidot has seemingly forgotten to do all over its codebase.

Reply:
I thought about this some more and I think we'll try it your way. Translators can copy the English file and be able to do the same sort of thing, and an obvious placeholder string will almost certainly be reported to us where a bit of stray English may not.